instrumentation/src/com/intellij/rt/coverage/instrumentation/Instrumenter.java

```diff
@@ -132,9 +132,7 @@ public void removeLine(final int line) {
 
   public void visitSource(String source, String debug) {
     super.visitSource(source, debug);
-    if (myShouldCalculateSource) {
-      myProjectData.getOrCreateClassData(myClassName).setSource(source);
-    }
+    myProjectData.getOrCreateClassData(myClassName).setSource(source);
```

Comment:
why is it now unconditional?

Reply:
I need source file name for xml report. I decided that storing one string per class is cheap enough to make it unconditional. I can change it to `if (myShouldCalculateSource || myReportFormat == ReportFormat.XML) { ... }`.

Reply:
I would set `myShouldCalculateSource` to true if `myReportFormat == ReportFormat.XML`, otherwise I don't understand why you need the field at all. And I would actually expect that additional non-interned string for each class could lead to a memory problem